### PR TITLE
chore(broker): renamed payload to variables

### LINKED ISSUE #2194: Inconistent on variables concept for publishing message
Here it is still called payload https://github.com/zeebe-io/zeebe/blob/develop/clients/java/src/main/java/io/zeebe/client/api/commands/PublishMessageCommandStep1.java

### PR DESCRIPTION
**Pull request by miguelpires**
Not everything was renamed because there are still some meaningful changes that need to be done, like getting rid of the payload column family in the VariablesState (#2225). After that we can either have an issue to mop up remaining occurrences of the old naming/concept or just apply the boy scout rule and rename them whenever we stumble upon them.  

closes #2194, #2195, #2215

### CODE REVIEW

File: clients/go/commands/publishMessage_command.go [hunk 1]
```diff
@@ -16,121 +16,121 @@
...
 
 import (
-	"context"
-	"fmt"
-	"github.com/zeebe-io/zeebe/clients/go/pb"
-	"github.com/zeebe-io/zeebe/clients/go/utils"
-	"time"
+    "context"
+    "fmt"
+    "github.com/zeebe-io/zeebe/clients/go/pb"
```
Comment (Zelldon) [inline clients/go/commands/publishMessage_command.go]: @menski do we have to somehow specify a rule for formatting like whitespaces instead of tabs  or idk ? :man_shrugging:

File: gossip/src/main/resources/gossip-schema.xml [hunk 2]
```diff
@@ -57,7 +57,7 @@
       <field name="senderGossipHeartbeat" id="11" type="uint64"/>
       <field name="senderId" id="12" type="uint16"/>
       <data name="eventType" id="14" type="varDataEncoding"/>
-      <data name="payload" id="15" type="varDataEncoding"/>
+      <data name="variables" id="15" type="varDataEncoding"/>
```
Comment (Zelldon) [inline gossip/src/main/resources/gossip-schema.xml]: Ok here it is not necessary :sweat_smile: but since we remove this anyway just leave it as it is.

File: clients/go/commands/publishMessage_command.go (hunk 1 above)
Comment (menski) [reply clients/go/commands/publishMessage_command.go]: running `gofmt` generates the expected format

**Comment by menski** [comment]
Please make sure that we mark these commits as breaking change, as they impact users.

**Comment by Zelldon** [comment]
Yes that is just what I was going to say thanks :+1:

**Review by ChrisKujawa — CHANGES_REQUESTED** [review]
Thanks for doing this :+1: 
There are still some payload usages, we need to change (besides the fact that the VariablesState has a payload columnFamily - we could change that to `temporaryVariableStore`)

    * JobRecordValueImpl
   * WorkflowInstanceSubscriptionRecordValueImpl
   * JobBatchActivateProcessor
   * PublishMessageTest
   * MessageStateTest
  * CompleteJobTest
   * MessageStreamProcessorTesti
   * ActivateJobsCommandStep1
   * JobWorkerBuilderStep1
  * etc..

Please run find or something similar to find the other occurrences. Please also have a look into the samples.

Please rerun on all go files the `gofmt` cmd, to fix your formatting changes.

As Basti mentioned mark the PR or your commits with the `BREAKING_CHANGE` flag and summarize what has changed (like rename setPayload to setVariables or on the records, payload was also changed etc.).

**Comment by MiguelPires** [comment]
Thanks @Zelldon. I didn't rename everything because it seems like the old concept is still being used and I thought that it might be better to rename once we're sure everything can be renamed. Also, I did search for usages of payload and but some actually make sense to me. For instance, one of the files you mentioned (JobWorkerBuilderStep1) has something like "(...) only contain variables from this list in their payload." which I would argue is actually correct. I can reword it to not use "payload" but it doesn't sound wrong to me as it is. In any case, I'll change all of it besides VariablesState.

**Comment by MiguelPires** [comment]
Hey @Zelldon, I addressed your comments. The things that weren't changed are:
* In the gossip module (since it will be deleted)
* Stuff related to raft/transport where payload isn't a payload in the "Zeebe" sense and more like a message payload
* Stuff related to the VariablesState payload/mapping thing (which according to https://github.com/zeebe-io/zeebe/issues/2225#issuecomment-475579059 doesn't need to be changed)

File: broker-core/src/main/java/io/zeebe/broker/workflow/state/EventScopeInstanceState.java [hunk 3]
```diff
@@ -170,7 +170,7 @@ private void createTrigger(
    * Returns the next event trigger for the event scope or null if none exists. This will not remove
    * the event trigger from the state.
    *
-   * @param eventScopeKey the key of the event scope
+   * @param eventScopeKey the key of the event scopen
```
Comment (Zelldon) [inline broker-core/src/main/java/io/zeebe/broker/workflow/state/EventScopeInstanceState.java]: I assume this was an mistake :)

Comment (Zelldon) [inline broker-core/src/main/java/io/zeebe/broker/workflow/state/EventScopeInstanceState.java]:
```suggestion
   * @param eventScopeKey the key of the event scope
```

**Review by ChrisKujawa — CHANGES_REQUESTED** [review]
Sorry for the late response. I added an commit to remove some more payload occurences, please also see the other comments.

File: gateway/pom.xml [hunk 4]
```diff
@@ -53,11 +55,6 @@
       <artifactId>msgpack-core</artifactId>
     </dependency>
 
-    <dependency>
-      <groupId>io.zeebe</groupId>
-      <artifactId>zb-msgpack-core</artifactId>
```
[the remark this answers is not in the document]
Comment (MiguelPires) [reply gateway/pom.xml]: No, I deleted the jmh folder at Menski's request so that dependency is no longer needed

[the remark this answers is not in the document]
Comment (Zelldon) [reply gateway/pom.xml]: ok thanks

**Comment by Zelldon** [comment]
I rebased the PR to the latest develop. Please squash the commits and write the BREAKING CHANGE message in the commit message. After that I will accept the PR

**Comment by MiguelPires** [comment]
Ok, thanks. I squashed and added the breaking change flag. Please let me know if it should be more descriptive or if it has any issues

**Review by ChrisKujawa — APPROVED** [review]
bors r+
